File: Haskell/CalcStats/test_CalcStats.hs
module Test_CalcStats where

import Test.HUnit
import CalcStats

testList = [6,9,15,(-2),92,11]
average_answer = 21.833333333333332

test_minimum_value_returns_negative_two =
  TestCase (assertEqual "The minimum value in the list is -2" (-2) (minimumValue testList))

test_maximum_value_returns_ninety_two =
  TestCase (assertEqual "The maximum value in the list is 92" 92 (maximumValue testList))

test_number_of_elements_is_six =
  TestCase (assertEqual "Number of elements is 6" 6 (numberOfElements testList))

test_average_is_correct =
  TestCase (assertEqual "Average is correct" average_answer (average testList))

tests = TestList [test_minimum_value_returns_negative_two,
                  test_maximum_value_returns_ninety_two,
                  test_number_of_elements_is_six,
                  test_average_is_correct]

main = do runTestTT tests

File: Ruby/CalcStats_TDD/test_calc_stats.rb
require './calc_stats'
require 'test/unit'

class TestCalcStats < Test::Unit::TestCase
  @@test_list = [6, 9, 15, -2, 92, 11]

  def test_the_minimum_value_of_a_test_list_is_negative_two
    answer = CalcStats.min_value(@@test_list)
    expected = -2

    assert_equal expected, answer
  end

  def test_the_maximum_value_of_a_test_list_is_ninety_two
    answer = CalcStats.max_value(@@test_list)
    expected = 92

    assert_equal expected, answer
  end

  def test_the_number_of_elements_in_test_list_is_six
    answer = CalcStats.number_of_elements(@@test_list)
    expected = 6

    assert_equal expected, answer
  end

  def test_the_average_is_within_some_small_delta
    answer = CalcStats.average(@@test_list)
    expected = 21.833333

    assert_equal true, within_delta(expected, answer, 0.000001)
  end

  def within_delta expected, answer, delta
    (answer - expected).abs < delta
  end

end
